source/base/mpi.cc

```diff
@@ -295,6 +297,81 @@ namespace Utilities
 
 
 
+    void
+    create_mpi_data_type_n_bytes(MPI_Datatype &result, std::size_t n_bytes)
+    {
+#  ifdef DEAL_II_WITH_MPI
+      // Simplified version from BigMPI repository, see
+      // https://github.com/jeffhammond/BigMPI/blob/5300b18cc8ec1b2431bf269ee494054ee7bd9f72/src/type_contiguous_x.c#L74
+      // (code is MIT licensed)
+
+      // We create an MPI datatype that has the layout A*nB where A is
+      // max_signed_int bytes repeated n times and B is the remainder.
+
+      const MPI_Count max_signed_int = (1U << 31) - 1;
```

Comment:
Isn't this `std::numeric_limits<MPI_Count>::max()`?

Reply:
no, MPI_Count is 64 bits. I could write this as ``numeric_limits<int>`` if you prefer.

Reply:
Ah, I see. Yes, let's make it `numeric_limits<int>` then.